File: python-unittest/tests/varar/yahtzee.steps.py
from varar import steps
from yahtzee_example import score

param, stimulus, sensor = steps()


@sensor("Examples of dice, category and score")
def _(state, row):
    dice = [int(d.strip()) for d in row["dice"].split(",")]
    return {"score": score(dice, row["category"])}

File: ruby-minitest/test/varar/yahtzee.steps.rb
# frozen_string_literal: true

require 'varar'
require_relative '../../lib/yahtzee'

steps do
  sensor('Examples of dice, category and score') do |_state, row|
    dice = row['dice'].split(',').map { |d| d.strip.to_i }
    { 'score' => Yahtzee.score(dice, row['category']) }
  end
end
